pkg/rancher-desktop/config/__tests__/settings.spec.ts

```diff
@@ -72,10 +249,255 @@ describe('updateFromCommandLine', () => {
+            } else if (inputPath.endsWith('locked.plist')) {
+              return lockedPlistProfile;
+            }
+          }
+          throw new Error("Shouldn't get here.");
+        };
+      }
+
+      describe('when there is no profile', () => {
+        beforeEach(() => {
+          mock = jest.spyOn(fs, 'readFileSync')
+            .mockImplementation(createMocker(0, 0));
+        });
+        afterEach(() => {
+          mock.mockRestore();
+        });
+        test('all fields are unlocked', () => {
+          settings.load();
+          verifyAllFieldsAreUnlocked(settings.getLockedSettings());
+        });
+      });
+
+      describe('when there is only a user profile with unlocked imageList', () => {
+        beforeEach(() => {
+          mock = jest.spyOn(fs, 'readFileSync')
+            .mockImplementation(createMocker(0, 1));
+        });
+        afterEach(() => {
+          mock.mockRestore();
+        });
+        test('all fields are unlocked', () => {
+          settings.load();
+          verifyAllFieldsAreUnlocked(settings.getLockedSettings());
+        });
+      });
+
+      describe('when there is only a user profile with locked imageList', () => {
+        beforeEach(() => {
+          mock = jest.spyOn(fs, 'readFileSync')
+            .mockImplementation(createMocker(0, 2));
+        });
+        afterEach(() => {
+          mock.mockRestore();
+        });
+        test('all fields are locked', () => {
+          settings.load();
+          verifyAllFieldsAreLocked(settings.getLockedSettings());
+        });
+      });
+
+      describe('when there is only a system profile with unlocked imageList', () => {
+        beforeEach(() => {
+          mock = jest.spyOn(fs, 'readFileSync')
+            .mockImplementation(createMocker(1, 0));
+        });
+        afterEach(() => {
+          mock.mockRestore();
+        });
+        test('all fields are unlocked', () => {
+          settings.load();
+          verifyAllFieldsAreUnlocked(settings.getLockedSettings());
+        });
+      });
+
+      describe('when both profiles exit, both with unlocked imageList', () => {
+        beforeEach(() => {
+          mock = jest.spyOn(fs, 'readFileSync')
+            .mockImplementation(createMocker(1, 1));
+        });
+        afterEach(() => {
+          mock.mockRestore();
+        });
+        test('all fields are locked', () => {
+          settings.load();
+          verifyAllFieldsAreUnlocked(settings.getLockedSettings());
+        });
+      });
+
+      describe('when the system profile is unlocked and the user profile is locked', () => {
+        beforeEach(() => {
+          mock = jest.spyOn(fs, 'readFileSync')
+            .mockImplementation(createMocker(1, 2));
+        });
+        afterEach(() => {
+          mock.mockRestore();
+        });
+        test('all fields are locked', () => {
+          settings.load();
+          verifyAllFieldsAreUnlocked(settings.getLockedSettings());
```

Comment:
This test might be backwards? If the user is locking things, why is the result unlocked?

Reply:
see other comment on generating code from comments (and not paying attention to comments when copying/pasting code 5 times)